### PR TITLE
Adds annotation support to stack.yml schema

### PR DESCRIPTION
**Pull request by ewilde**
## Description
Add annotation support the stack.yml schema to enable metadata for functions which may be used by the back-end for management, orchestration, events and build tasks.
	
- [x] I have raised an issue to propose this change: https://github.com/openfaas/faas/issues/682

## How Has This Been Tested?
This has been tested using the following combinations:

1. All components supporting annotations: new cli -> new gateway -> new faas-netes
2. Some components not support annotation: old cli -> new gateway -> new faas-netes
3. Some components not support annotation: old cli -> old gateway -> new faas-netes

For more details see: https://github.com/openfaas/faas-netes/pull/258

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 [[constraint]]
   name = "github.com/openfaas/faas"
-  version = "0.6.16"
+  version = "0.8.8"
```
Comment (ewilde) [inline Gopkg.toml]: waiting for new `faas` release

Comment (alexellis) [reply Gopkg.toml]: Thanks, it's now released and available on GitHub.

File: commands/deploy.go [hunk 2]
```diff
@@ -212,7 +213,29 @@ Error: %s`, fprocessErr.Error())
...
 				Requests: function.Requests,
 			}
-			statusCode := proxy.DeployFunction(function.FProcess, services.Provider.GatewayURL, function.Name, function.Image, function.RegistryAuth, function.Language, deployFlags.replace, allEnvironment, services.Provider.Network, functionConstraints, deployFlags.update, deployFlags.secrets, allLabels, functionResourceRequest1)
+
+			var annotations map[string]string
+			if function.Annotations != nil {
+				annotations = *function.Annotations
+			}
+
+			statusCode := proxy.DeployFunction(
```
Comment (alexellis) [inline commands/deploy.go]: This got a bit too long around the time we added "Language" and now it just got annotations plus the read-only flag. Probably time we refactored.. please could you raise an issue to refactor this into a `DeployFunctionStruct`?

Comment (alexellis) [reply commands/deploy.go]: cc @LucasRoesler

File: commands/deploy.go [hunk 3]
```diff
@@ -276,7 +299,7 @@ func deployImage(
 		image, registryAuth, language,
 		deployFlags.replace, envvars, network,
 		deployFlags.constraints, deployFlags.update, deployFlags.secrets,
-		labelMap, functionResourceRequest1)
+		labelMap, nil, functionResourceRequest1)
```
Comment (alexellis) [inline commands/deploy.go]: What about passing an empty version of this instead or a named nil variable? (for readability)

**Comment by alexellis** [comment]
Hi @ewilde unfortunately we have had some PRs touching the same areas. This will need a rebase. Please don't do a `git merge`.

Alex

File: commands/deploy.go (hunk 2 above)
Comment (LucasRoesler) [reply commands/deploy.go]: I agree that we should do this, I was actually thinking about this in my PR but didn't want to introduce a change like that at the same time as the read only behavior 👍

Comment (alexellis) [reply commands/deploy.go]: @LucasRoesler agreed, it is best to work in increments. Please could you raise an issue for it today? 🐞

**Comment by alexellis** [comment]
LGTM

File: commands/deploy.go (hunk 2 above)
Comment (LucasRoesler) [reply commands/deploy.go]: done in #475

**Pull request merged**